scripts/tests/providers-directive-guard.test.js

```
@@ -0,0 +1,149 @@
+    }
+
+    expect(violations, violations.join('\n')).toEqual([]);
+  });
+
+  it('eslint.config.js has no provider-specific central exclusions or rule-off blocks', () => {
+    const configContent = readFileSync(ESLINT_CONFIG_PATH, 'utf8');
+    const violations = [];
+    const providerPathPattern = new RegExp(PROVIDER_PATH, 'g');
+    let match;
+
+    while ((match = providerPathPattern.exec(configContent)) !== null) {
```

Biome (2.5.0)

[error] The assignment should not be in an expression.

(lint/suspicious/noAssignInExpressions)